pkg/engine/internal/executor/executor.go

```diff
@@ -16,10 +16,17 @@ import (
 	"github.com/grafana/loki/v3/pkg/dataobj/sections/logs"
 	"github.com/grafana/loki/v3/pkg/dataobj/sections/streams"
 	"github.com/grafana/loki/v3/pkg/engine/internal/planner/physical"
+	"github.com/grafana/loki/v3/pkg/xcap"
 )
 
 var tracer = otel.Tracer("pkg/engine/internal/executor")
 
+// scopeName returns a unique name for a scope based on the node type and ID.
+// This ensures that each node, including partitioned nodes, has a unique scope name.
+func scopeName(node physical.Node) string {
+	return fmt.Sprintf("%s-%s", node.Type().String(), node.ID().String())
+}
```

Comment:
I'm not sure about this one; it's going to make it more difficult for us to examine captures over time when we export to other telemetry types: for example, we couldn't export as profiles and see how much time was spent in DataObjScan over the last 12h, because each region is being given a unique name.

I suspect we'll want region name cardinality to be fairly low. Instead, this seems like a more appropriate use for attributes to add extra information; can you add the node ID as an attribute and use that for the enhanced printed form?